File: dotnet/Lexy.Poc/Lexy.Poc.Core/Language/EnumName.cs
using System.Collections.Generic;
using Lexy.Poc.Core.Parser;
using Microsoft.CodeAnalysis.CSharp;

namespace Lexy.Poc.Core.Language
{
    public class EnumName : Node
    {
        public string Value { get; private set; }

        public EnumName(SourceReference sourceReference) : base(sourceReference)
        {
        }

        public void ParseName(string parameter)
        {
            Value = parameter;
        }

        protected override IEnumerable<INode> GetChildren()
        {
            yield break;
        }

        protected override void Validate(IValidationContext context)
        {
            if (string.IsNullOrEmpty(Value))
            {
                context.Logger.Fail(Reference, $"Invalid enum name: {Value}. Name should not be empty.");
            }
            if (!SyntaxFacts.IsValidIdentifier(Value))
            {
                context.Logger.Fail(Reference, $"Invalid enum name: {Value}.");
            }
        }
    }
}

File: typescipt/src/language/enums/enumName.ts


export class EnumName extends Node {
   public string Value { get; private set; }

   public EnumName(SourceReference sourceReference) : base(sourceReference) {
   }

   public parseName(parameter: string): void {
     Value = parameter;
   }

   public override getChildren(): Array<INode> {
     yield break;
   }

   protected override validate(context: IValidationContext): void {
     if (string.IsNullOrEmpty(Value))
       context.logger.fail(this.reference, $`Invalid enum name: {Value}. Name should not be empty.`);
     if (!SyntaxFacts.IsValidIdentifier(Value)) context.logger.fail(this.reference, $`Invalid enum name: {Value}.`);
   }
}
